### PR TITLE
F 20 e iam versioning

### PR DESCRIPTION
**Pull request by archanabpatil**
TA for 4 tests under https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/980255029/F-20E+S3+API+IAM+Policy+Support+-+PI7
[TEST-46505](https://jts.seagate.com/browse/TEST-46505)
[TEST-46504](https://jts.seagate.com/browse/TEST-46504)
[TEST-46503](https://jts.seagate.com/browse/TEST-46503)
[TEST-46501](https://jts.seagate.com/browse/TEST-46501)
Jira story : https://jts.seagate.com/browse/CORTX-33471

### CODE REVIEW

**Comment by ketanarlulkar** [comment]
@archanabpatil, please execute tests on Ceph and Cortx and attach execute logs. Also add CORTX JIRA story to the PR

File: s3tests_boto3/functional/test_iam.py [hunk 1]
```diff
@@ -1795,3 +1796,290 @@ def test_copy_object_allow_get_src_put_dest():
...
+                           PolicyName='deny_policy_versioning', UserName=get_iam_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+    # Perform Get Bucket versioning operation on  “iam1buk1”
+    response = s3_client_iam.get_bucket_versioning(Bucket=bucket)
+    e = assert_raises(ClientError, s3_client_iam.get_bucket_versioning, Bucket=bucket)
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 403)
+    eq(error_code, 'AccessDenied')
+    #Perform Put Bucket versioning on “iam1buk1”
```
Comment (ketanarlulkar) [inline s3tests_boto3/functional/test_iam.py]: Use neutral quotation marks
```suggestion
    #Perform Put Bucket versioning on "iam1buk1"
```

Comment (archanabpatil) [reply s3tests_boto3/functional/test_iam.py]: agreed

Comment (apoorva-rao) [reply s3tests_boto3/functional/test_iam.py]: @archanabpatil let's mark this as resolved once the changes suggested by @ketanarlulkar have been incorporated

File: s3tests_boto3/functional/test_iam.py [hunk 2]
```diff
@@ -1795,3 +1796,290 @@ def test_copy_object_allow_get_src_put_dest():
...
+@attr(operation='Verify Allow and Deny Put Bucket Versioning API using IAM policy for self')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_put_bucket_versioning_iam_policy():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    bucket = get_new_bucket(client=s3_client_iam, name="iam1buk1")
+    s3_client_iam.put_object(Bucket=bucket, Key='iam1buk1obj1', Body='bar')
+    # Perform Put Bucket Versioning operation on  “iam1buk1” with  “Enabled”
```
Comment (apoorva-rao) [inline s3tests_boto3/functional/test_iam.py]: Same as @ketanarlulkar's comment at 1835. Please make this correction across other comments as well in this commit

File: s3tests_boto3/functional/test_iam.py [hunk 3]
```diff
@@ -1795,3 +1796,290 @@ def test_copy_object_allow_get_src_put_dest():
...
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow and Deny Get Bucket Versioning API using IAM policy for self')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_get_bucket_versioning_iam_policy():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    bucket = get_new_bucket(client=s3_client_iam, name="iam1buk1")
```
Comment (ketanarlulkar) [inline s3tests_boto3/functional/test_iam.py]: Don't give bucket name, let it autogenerate

```suggestion
    bucket = get_new_bucket(client=s3_client_iam)
```

File: s3tests_boto3/functional/test_iam.py [hunk 4]
```diff
@@ -1795,3 +1796,290 @@ def test_copy_object_allow_get_src_put_dest():
...
+    s3_client_iam.put_object(Bucket=bucket, Key='iam1buk1obj1', Body='bar')
+    # Create IAM user “iam2”
+    s3_client_alt = get_alt_client()
+    bucket2 = get_new_bucket(client=s3_client_alt, name="iam2buk1")
+    s3_client_alt.put_object(Bucket=bucket, Key='iam2buk1obj1', Body='foobar')
```
Comment (ketanarlulkar) [inline s3tests_boto3/functional/test_iam.py]: Do you mean bucket2
```suggestion
    s3_client_alt.put_object(Bucket=bucket2, Key='iam2buk1obj1', Body='foobar')
```

File: s3tests_boto3/functional/test_iam.py (hunk 1 above)
Comment (ketanarlulkar) [reply s3tests_boto3/functional/test_iam.py]: This is there in multiple places

File: s3tests_boto3/functional/test_iam.py (hunk 4 above)
Comment (archanabpatil) [reply s3tests_boto3/functional/test_iam.py]: modified

**Comment by archanabpatil** [comment]
https://github.com/Seagate/s3-tests/pull/22/files is the correct PR to be reviewed @ketanarlulkar  @apoorva-rao  @ravindrachoudhari

**Pull request closed without merging**